src/authentication/authentication.handler.ts

```diff
@@ -9,29 +9,46 @@ import * as jwt from 'jsonwebtoken';
 import { ApplicationError } from '../utils/errors/applicationError';
 import { readFile } from 'fs';
 
-export class AuthenticationHandler {
+const ShragaStrategy = require('passport-shraga').Strategy;
 
+export class AuthenticationHandler {
     static initialize(app: Application) {
         app.use(passport.initialize());
+        app.use(passport.session());
 
-        passport.serializeUser(AuthenticationHandler.serialize);
-        passport.deserializeUser(AuthenticationHandler.deserialize);
-
-        passport.use(new Strategy(
-            config.authentication.saml as SamlConfig,
-            AuthenticationHandler.verifyUser,
-        ));
+        passport.serializeUser(this.serialize);
+        passport.deserializeUser(this.deserialize);
 
+        this.configurePassport();
         return passport.initialize();
     }
 
+    protected static serialize(user: { id: string }, done: (err?: Error, id?: string) => void) {
+    }
+
+    protected static async deserialize(id: string, done: (err?: Error, user?: any) => void) {
+    }
+
+    protected static configurePassport() {
+    }
+
     static handleUser(req: Request, res: Response) {
-        const userToken = jwt.sign(req.user, config.authentication.secret);
+        const minute = 60;
+        const hour = 60 * minute;
+        const day = 24 * hour;
+        const expiresIn = day * config.authentication.daysExpires;
+        const user: Object = Object.assign(
```

Comment:
Using `Object.assign` is an old way of doing this thing, you could do this in a more syntactically clear way with spread operator like `const user: Object = { exp: Math.floor(Date.now() / 1000) + expiresIn, ...req.user };`.
There are other lines in various services you've been working on where you used `Object.assign` and could rewrite it with spread operator.